### PR TITLE
Chore/mf 1902 use Backbase BOM

### PR DESCRIPTION
**Pull request by WingEraser**
Use Backbase BOM 2024.07.01
Java 17.
Gradle 8.7.
Third party libraries are updated. They are on separate Version Catalog (for now).
Authentication Journey got affected most.

### CODE REVIEW

File: accounts-journey/build.gradle.kts [hunk 1]
```diff
@@ -10,8 +10,13 @@ android {
...
+
+    androidTestImplementation(libs.navigation.testing)
+
     // Backbase libraries
+    implementation(backbase.bom)
     implementation(backbase.bundles.common)
-    implementation(libs.bundles.navigation)
-    androidTestImplementation(project("path" to ":fake-accounts-use-case"))
+
+    androidTestImplementation(project(":fake-accounts-use-case"))
```
Comment (matiasdelbel) [inline accounts-journey/build.gradle.kts:21]: you can opt-in for `TYPESAFE_PROJECT_ACCESSORS` feature in your `settings.gradle` to use a type safe project names:

```kotlin
# settings.gradle file
enableFeaturePreview("TYPESAFE_PROJECT_ACCESSORS")
```

Then, in your `build.gradle.file`

```kotlin
// replace this line 
androidTestImplementation(project(":fake-accounts-use-case"))

// with this
androidTestImplementation(projects.fakeAccountsUseCase)
```

File: accounts-use-case/src/main/res/values/colors.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <color name="white">#FFFFFF</color>
```
Comment (matiasdelbel) [inline accounts-use-case/src/main/res/values/colors.xml:3]: is this color needed?

File: buildSrc/src/main/kotlin/base-android-library-module.gradle.kts [hunk 3]
```diff
@@ -27,11 +27,11 @@ android {
         }
     }
     compileOptions {
-        sourceCompatibility = JavaVersion.VERSION_11
-        targetCompatibility = JavaVersion.VERSION_11
+        sourceCompatibility = JavaVersion.VERSION_17
+        targetCompatibility = JavaVersion.VERSION_17
     }
     kotlinOptions {
```
Comment (matiasdelbel) [inline buildSrc/src/main/kotlin/base-android-library-module.gradle.kts:33]: nit: kotlinOptions will be deprecated in kotlin 2.0, the alternative is already available:

```kotlin
kotlin {
    compilerOptions { jvmTarget.set(JvmTarget.JVM_17) }
}
```

File: gradle/backbase.versions.toml [hunk 4]
```diff
@@ -1,75 +1,56 @@
...
+more-menu-journey = "6.1.0"
 
 ## W
-workspaces-journey = "6.0.0"
+workspaces-journey = "7.0.0"
 
 [libraries]
 ## A
-access-control-client-2-entitlements-use-case = { group = "com.backbase.android.retail.feature_filter", name = "accesscontrol-client-2-entitlements-use-case", version.ref = "access-control-client-2-entitlements-use-case" }
+access-control-client-entitlements-use-case = { group = "com.backbase.android.retail.feature_filter", name = "accesscontrol-client-2-entitlements-use-case", version.ref = "access-control-client-2-entitlements-use-case" }
```
Comment (matiasdelbel) [inline gradle/backbase.versions.toml:20]: I wonder if we need to add this use case to the shared catalog + BOM

File: gradle/libs.versions.toml [hunk 5]
```diff
@@ -1,14 +1,14 @@
 [versions]
```
Comment (matiasdelbel) [inline gradle/libs.versions.toml:1]: Some of this third parties libraries are also available in the BOM. Check the catalog definition to see which one: https://github.com/backbase-rnd/mobile-foundation-dependencies/blob/master/catalog/third-parties/gradle/libs.versions.toml

**Review by matiasdelbel — APPROVED** [review]

**Review by charbelmkh — APPROVED** [review]

**Review by charbelmkh — APPROVED** [review]

**Pull request merged**